src/ride/track.c

```diff
@@ -370,7 +370,7 @@ void track_list_populate(ride_list_item item, uint8* track_list_cache){
 		}
 
 		// If cur_track_entry_index is greater than max number of tracks
-		if (cur_track_entry_index >= 1000){
+		if (cur_track_entry_index >= 254){
```

Comment:
Other than this, does the rest of this commit look good for merging? I can revert this line and create a separate issue for it.